backends/platform/ios7/ios7_video.mm

```diff
@@ -228,6 +230,23 @@ - (void)setupGestureRecognizers {
 	[twoFingerLongPressGesture setDelaysTouchesEnded:NO];
 	[twoFingerLongPressGesture setCancelsTouchesInView:NO];
 	[twoFingerLongPressGesture canPreventGestureRecognizer:twoFingerTapGesture];
+	
+	pencilTouchGesture = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(pencilTouch:)];
+	[pencilTouchGesture setNumberOfTouchesRequired:1];
+	[pencilTouchGesture setAllowedTouchTypes:@[@(UITouchTypePencil)]];
+	[pencilTouchGesture setMinimumPressDuration:0];
```

Comment:
If setting the minimum time to 0.5 seconds here you could act on the gesture begin event and not keep track of the time yourself.

Reply:
Thanks for your comments.

The issue I ran into with this was in having to use `UILongPressGestureRecognizer` to detect a regular touch of the Pencil. If I set the minimum duration to anything other than a near instant response to your touch then it ends up feeling "bad" to actually use. Because this regular touch "long" gesture is then active whenever you touch the screen, I found that any separate long touch gesture would never be recognised - it seems like only one of this gesture type can be recognised at once.

I tried a few variants of this where the gestures were split out into their own individual gestures, like the touches are, but I couldn't find a way to make them work reliably. For example I put the long touch on a double tap and hold gesture, and while that worked in many instances it failed in others like the COMI inventory screen - where 9/10 times you end up instantly selecting the item in the inventory that you wanted to long hold on, because this touch gesture was always winning.

So in the end I resorted to combining the regular touch and long touch gestures into one and tracking how long it's been pressed manually.